dataactvalidator/config/sqlrules/d23_detached_award_financial_assistance.sql

```diff
@@ -0,0 +1,13 @@
+-- AwardingSubTierAgencyCode is required for all submissions and cannot be blank.
+-- AwardingSubTierAgencyCode must contain a valid four character numeric code.
+SELECT
+    dafa.row_number,
+    dafa.awarding_sub_tier_agency_c
+FROM detached_award_financial_assistance as dafa
+WHERE dafa.submission_id = {0}
+    AND (dafa.awarding_sub_tier_agency_c != ''
```

Comment:
Also don't need this outer set. You probably also don't need to check for '' because that situation will cause an error earlier in the process anyway.